### PR TITLE
Feat/AN-4121 implement aliasby selection

### PR DESCRIPTION
**Pull request by MGJamJam**
Issue: https://bitmovin.atlassian.net/browse/AN-4121

Work: 
- Implements Alias by selection
- ignores disabled Queries
image

image

### CODE REVIEW

File: bitmovin-analytics-datasource/src/components/QueryEditor.tsx [hunk 1]
```diff
@@ -102,6 +102,11 @@ export function QueryEditor({ query, onChange, onRunQuery, datasource }: Props)
     onRunQuery();
   };
 
+  const onAliasByChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onChange({ ...query, aliasBy: event.target.value });
+    onRunQuery();
```
Comment (sabe25) [inline bitmovin-analytics-datasource/src/components/QueryEditor.tsx:107]: Is it necessary to run the query again just for changing the alias? It is not part of the query right?

File: bitmovin-analytics-datasource/src/types.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ export interface BitmovinAnalyticsDataQuery extends DataQuery {
   orderBy: QueryOrderBy[];
   limit?: number;
   filters: QueryFilter[];
+  aliasBy?: string;
```
Comment (sabe25) [inline bitmovin-analytics-datasource/src/types.ts:24]: Why are you adding this field to the query?
According to our api there is no aliasBy on the query

Comment (MGJamJam) [reply bitmovin-analytics-datasource/src/types.ts:24]: This is the DataQuery interface, not the RequestQuery Interface. So this interface defines the types that are possible to set in the QueryEditor and get passed to the query method, where the actual RequestQuery is then built.

**Review by sabe25 — APPROVED** [review]

File: bitmovin-analytics-datasource/src/components/QueryEditor.tsx (hunk 1 above)
Comment (MGJamJam) [reply bitmovin-analytics-datasource/src/components/QueryEditor.tsx:107]: I just tested it. Yes it is not part of the query put if we don't call the onRunQuery then the query method does not get called which also means that the alias does not get applied. So we need to call it here, even though we wouldn't need to query our API. Maybe we could check in the query method what attribute changed and not query our API when only the alias by changes, but honestly I don't think this is worth the effort 🤔

File: bitmovin-analytics-datasource/src/components/QueryEditor.tsx [hunk 3]
```diff
@@ -102,6 +102,11 @@ export function QueryEditor({ query, onChange, onRunQuery, datasource }: Props)
     onRunQuery();
   };
 
+  const onAliasByChange = (event: ChangeEvent<HTMLInputElement>) => {
```
Comment (Svarozic) [inline bitmovin-analytics-datasource/src/components/QueryEditor.tsx:105]: nit: event `onBlur` should be handled by handler, aka `handleAliasByBlur`

**Review by Svarozic — APPROVED** [review]

**Pull request merged**